### PR TITLE
fix(steps): handle cols with all nulls in `DropZeroVariance`

### PR DESCRIPTION
**Pull request by jitingxu1**
add check to see if `var`of a numeric column is `nan`

### CODE REVIEW

File: ibis_ml/steps/_select_features.py [hunk 1]
```diff
@@ -62,16 +63,16 @@ def fit_table(self, table: ir.Table, metadata: Metadata) -> None:
...
-                c = table[name]
-                if isinstance(c, ir.NumericColumn):
+            for col_name in columns:
+                col_var = results[f"{col_name}_var"]
+                if isinstance(table[col_name], ir.NumericColumn):
                     # Check variance for numeric columns
-                    if results[f"{name}_var"] < self.tolerance:
-                        cols.append(name)
-                elif results[f"{name}_var"] < 2:
+                    if not col_var or math.isnan(col_var) or col_var < self.tolerance:
```
Comment (deepyaman) [inline ibis_ml/steps/_select_features.py:70]: What is `not col_var` for?

File: ibis_ml/steps/_select_features.py [hunk 2]
```diff
@@ -62,16 +63,16 @@ def fit_table(self, table: ir.Table, metadata: Metadata) -> None:
...
+                        cols.append(col_name)
+                elif not col_var or col_var < 2:
```
Comment (deepyaman) [inline ibis_ml/steps/_select_features.py:72]: Same here.

**Review by deepyaman — COMMENTED** [review]
Looks good overall, just not sure I understand one thing.

File: ibis_ml/steps/_select_features.py (hunk 1 above)
Comment (deepyaman) [reply ibis_ml/steps/_select_features.py:70]: Can you add a test for `None` then? Also, if testing for `None`, should be `col_var is None` instead of `not col_var`.

Comment (deepyaman) [reply ibis_ml/steps/_select_features.py]: `col_var` for the all-`None` column is 0, so this isn't getting hit. I don't know if it's reachable some other way?

File: ibis_ml/steps/_select_features.py [hunk 3]
```diff
@@ -62,16 +63,16 @@ def fit_table(self, table: ir.Table, metadata: Metadata) -> None:
...
+            for col_name in columns:
+                col_var = results[f"{col_name}_var"]
+                if isinstance(table[col_name], ir.NumericColumn):
                     # Check variance for numeric columns
-                    if results[f"{name}_var"] < self.tolerance:
-                        cols.append(name)
-                elif results[f"{name}_var"] < 2:
+                    if math.isnan(col_var) or col_var < self.tolerance:
+                        cols.append(col_name)
+                elif col_var is None or col_var < 2:
```
Comment (deepyaman) [inline ibis_ml/steps/_select_features.py:72]:
```suggestion
                elif col_var < 2:
```

**Review by deepyaman — APPROVED** [review]
LGTM! Thanks for indulging the back and forth
